graviti/manager/branch.py

```diff
@@ -3,19 +3,59 @@
 # Copyright 2022 Graviti. Licensed under MIT License.
 #
 
-"""The implementation of the BranchManager."""
+"""The implementation of the Branch and BranchManager."""
 
-from typing import TYPE_CHECKING, Generator, Optional
+from functools import partial
+from typing import TYPE_CHECKING, Dict, Generator, Optional
 
 from tensorbay.client.lazy import PagingList
-from tensorbay.client.struct import Branch
+from tensorbay.client.struct import Branch as TensorbayBranch
+from tensorbay.utility import attr
 
 from graviti.client import list_branches
 from graviti.exception import ResourceNotExistError
+from graviti.manager.commit import ROOT_COMMIT_ID, NamedCommit
 
 if TYPE_CHECKING:
     from graviti.manager.dataset import Dataset
 
+_ERROR_MESSAGE = "The '{attr_name}' is not available due to this branch has no commit history."
+_attr = partial(attr, is_dynamic=True, error_message=_ERROR_MESSAGE)
+
+
+class Branch(NamedCommit):
+    """This class defines the structure of a branch.
+
+    Arguments:
+        name: The name of the branch.
+        commit_id: The commit id.
+        parent_commit_id: The parent commit id.
+        title: The commit title.
+        description: The commit description.
+        committer: The commit user.
+        committed_at: The time when the draft is committed.
+
+    """
+
+    parent_commit_id: str = _attr()
+    title: str = _attr()
+    description: str = _attr()
+    committer: str = _attr()
+    committed_at: str = attr()
+
+    def _loads(self, contents: Dict[str, str]) -> None:
```

Comment:
We should not use `load` anymore, @Lee-000 will help on removing the loads and dumps in the future.

Reply:
Because this _load is associated with common_load, we need another pr to modify them.